### PR TITLE
Bug fix : Add version in the error messages for plugin lifecycle methods

### PR DESCRIPTION
**Pull request by mpanchajanya**
### What this PR does / why we need it

- Improve the error message of the `tanzu plugin sync` command to mention the version of the plugin if the plugin is not found.

*Previously*

```
~ $ tanzu plugin list
Standalone Plugins
  NAME     DESCRIPTION                 TARGET      VERSION                 STATUS
  builder  Build Tanzu components      global      v0.1.0-dev-8-g6087ea00  installed
  apps     Applications on Kubernetes  kubernetes  v0.10.0                 installed

Plugins from Context:  k8s-cluster-1
  NAME                DESCRIPTION                    TARGET      VERSION  STATUS
  cluster             Kubernetes cluster operations  kubernetes  v0.28.0  installed
  kubernetes-release  Kubernetes release operations  kubernetes  v0.28.1  not installed

[!] As shown above, some recommended plugins have not been installed. To install them please run 'tanzu plugin sync'.

``` 

```
~ $ tanzu plugin sync
[i] Checking for required plugins...
[x] : unable to find plugin 'kubernetes-release' for target 'kubernetes'

```

*After the improvement with this fix*

Improve the error message to include the plugin version as well. Example:

```
[x] : unable to find plugin 'kubernetes-release' with version 'v0.28.1' for target 'kubernetes'

```

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

- Updated Unit tests to verify the error message to include plugin version
-  E2E test to verify the new error message that includes plugin version

```
mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (fix_plugin_sync_err_msg)> tanzu plugin search
  NAME                DESCRIPTION                  TARGET           LATEST
  isolated-cluster    Desc for isolated-cluster    global           v9.9.9
  pinniped-auth       Desc for pinniped-auth       global           v9.9.9
  cluster             Desc for cluster             kubernetes       v9.9.9
  feature             Desc for feature             kubernetes       v9.9.9
  kubernetes-release  Desc for kubernetes-release  kubernetes       v9.9.9
  management-cluster  Desc for management-cluster  kubernetes       v9.9.9
  package             Desc for package             kubernetes       v9.9.9
  secret              Desc for secret              kubernetes       v9.9.9
  telemetry           Desc for telemetry           kubernetes       v9.9.9
  account             Desc for account             mission-control  v9.9.9
  apply               Desc for apply               mission-control  v9.9.9
  audit               Desc for audit               mission-control  v9.9.9
  cluster             Desc for cluster             mission-control  v9.9.9
  clustergroup        Desc for clustergroup        mission-control  v9.9.9
  continuousdelivery  Desc for continuousdelivery  mission-control  v9.9.9
  data-protection     Desc for data-protection     mission-control  v9.9.9
  ekscluster          Desc for ekscluster          mission-control  v9.9.9
  events              Desc for events              mission-control  v9.9.9
  helm                Desc for helm                mission-control  v9.9.9
  iam                 Desc for iam                 mission-control  v9.9.9
  inspection          Desc for inspection          mission-control  v9.9.9
  integration         Desc for integration         mission-control  v9.9.9
  management-cluster  Desc for management-cluster  mission-control  v9.9.9
  policy              Desc for policy              mission-control  v9.9.9
  secret              Desc for secret              mission-control  v9.9.9
  tanzupackage        Desc for tanzupackage        mission-control  v9.9.9
  workspace           Desc for workspace           mission-control  v9.9.9
mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (fix_plugin_sync_err_msg)> tanzu plugin install kubernetes-release -v 1.1.1
[x] : unable to find plugin 'kubernetes-release' with version '1.1.1'
mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (fix_plugin_sync_err_msg) [1]> tanzu plugin install kubernetes-release -v 1.1.1 -t k8s
[x] : unable to find plugin 'kubernetes-release' with version '1.1.1' for target 'kubernetes'
mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (fix_plugin_sync_err_msg) [1]> tanzu plugin install kubernetes-release -v 1.1.1 -t tmc
[x] : unable to find plugin 'kubernetes-release' with version '1.1.1' for target 'mission-control'
mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (fix_plugin_sync_err_msg) [1]>
```

### Release note

```release-note
 plugin install/sync error message now include version of plugin it failed to install
```

### CODE REVIEW

**Review by chandrareddyp — APPROVED** [review]
LGTM

File: pkg/pluginmanager/manager_test.go [hunk 1]
```diff
@@ -268,7 +268,7 @@ func Test_InstallPlugin_InstalledPlugins_Central_Repo(t *testing.T) {
 	// Try installing the feature plugin which is targeted for k8s but requesting the TMC target
 	err = InstallStandalonePlugin("feature", "v0.2.0", configtypes.TargetTMC)
 	assertions.NotNil(err)
-	assertions.Contains(err.Error(), "unable to find plugin 'feature' for target 'mission-control'")
+	assertions.Contains(err.Error(), "unable to find plugin 'feature' with version 'v0.2.0' for target 'mission-control'")
```
Comment (chandrareddyp) [inline pkg/pluginmanager/manager_test.go:271]: nit, if possible, in this file only two versions are referred to, "v1.6.0" and "v0.2.0", please convert them to constants and replace in all places.

**Comment by anujc25** [comment]
So, if the plugin doesn't exist and when the user doesn't provide a `--version`, this change shows an error like the below which makes sense to me but I wanted to note it for awareness.
```
tz plugin install fakeplugin
[x] : unable to find plugin 'fakeplugin' with version 'latest'
```

**Review by anujc25 — APPROVED** [review]
LGTM. Thanks

**Pull request merged**